packages/castmill/lib/castmill/addons/devices/components/device-cache.tsx

```diff
@@ -0,0 +1,139 @@
+import { Component, createSignal } from 'solid-js';
+import { AiOutlineDelete } from 'solid-icons/ai';
+
+import {
+  Tabs,
+  TableView,
+  TableViewRef,
+  SortOptions,
+  Column,
+} from '@castmill/ui-common';
+
+import { Device } from '../interfaces/device.interface';
+
+import { DevicesService } from '../services/devices.service';
+
+interface DeviceTableCacheItem {
+  timestamp: string;
+  url: string;
+  size: number;
+  accessed: number;
+  mimeType: string;
+}
+
+const columns = [
+  { key: 'timestamp', title: 'Timestamp', sortable: true },
+  { key: 'url', title: 'Url', sortable: true },
+  { key: 'size', title: 'Size', sortable: false },
+  { key: 'accessed', title: 'Accessed', sortable: false },
+  { key: 'mimeType', title: 'Mime Type', sortable: false },
+] as Column<DeviceTableCacheItem>[];
+
+export const DeviceCache: Component<{ device: Device }> = (props) => {
+  const [selectedItems, setSelectedItems] = createSignal(new Set<string>());
+
+  const itemsPerPage = 10; // Number of items to show per page
+
+  const fetchCachePage = async (
+    type: 'data' | 'code' | 'medias',
+    {
+      page,
+      sortOptions,
+      search,
+      filters,
+    }: {
+      page: { num: number; size: number };
+      sortOptions: SortOptions;
+      search?: string;
+      filters?: Record<string, string | boolean>;
+    }
+  ) => {
+    return DevicesService.getDeviceCache(props.device.id, {
+      type,
+      page: page.num,
+      page_size: page.size,
+      sortOptions,
+    });
+  };
+
+  const onRowSelect = (rowsSelected: Set<string>) => {
+    setSelectedItems(rowsSelected);
+  };
+
+  let tableViewRef: TableViewRef;
+
+  const setRef = (ref: TableViewRef) => {
+    tableViewRef = ref;
+  };
+
+  const refreshData = () => {
+    if (tableViewRef) {
+      tableViewRef.reloadData();
+    }
+  };
+
+  const actions = [
+    {
+      icon: AiOutlineDelete,
+      handler: (item: DeviceTableCacheItem) => {
+        console.log('Delete device', item);
```

Comment:
How do we want to handle these kind of not-implemented-yet situations? Is a console.log enought, or should we add a //TODO-comment too? Or maybe even create an issue for it? TODO comment and a console.log maybe?

Reply:
The thing with the console.log is that it tells you that the handler is being called, thats why I like it, but we should definitely have issues so that we do not forget about them. I will create one for this right away.

Reply:
https://github.com/castmill/castmill/issues/39